mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1610,6 +1612,16 @@ void ModuleImport::processFunctionAttributes(llvm::Function *func,
     funcOp.setArmStreaming(true);
   else if (func->hasFnAttribute("aarch64_pstate_sm_body"))
     funcOp.setArmLocallyStreaming(true);
+  llvm::Attribute attr = func->getFnAttribute(llvm::Attribute::VScaleRange);
+  if (attr.isValid()) {
```

Comment:
Does `attr.isValid()` return if function even has this attribute? Above it looks like there is also a specific function to achieve this (`func->hasFnAttribute("vscale_range")`)?

Reply:
The reason the lines above use the `hasFnAttr` is that they don't actually need the attribute as such (in fact, they are renaming the attribute to a differnet name, and it's a unit attribute. I need to get the min/max values out of the attribute that came from LLVM. And yes, `isValid` returns the correct value (false) when there's no attribute - the returned attribute has a NULL `pImpl` if not present, which makes `isValid` say "not valid" - I manually tested with functions that haven't got any attribute - and of coruse, this particular section of code gets run pretty much any time there's a function in some MLIR file, so plenty of other tests will show that "not having an attribute does cause this to not fire. 

If you want me to add a test for "not having this attribute", I'll do so, but there's plenty of other tests that at least show "it doesn't cause a problem".

Reply:
Thanks for the explanation. All good, I was just curious how it works!